miner/block_builder.go

```diff
@@ -194,9 +195,17 @@ func (t *BlockBuilder) createBlock(id types.LayerID, atxID types.AtxId, eligibil
 				log.Uint64("bottom", uint64(bottom)), log.Uint64("top", uint64(top)),
 				log.Uint64("hdist", uint64(t.hdist)), log.Err(err))
 		}
-		if votes == nil { // if no votes set to empty
-			t.Info("Votes is nil. Setting votes to an empty array")
-			votes = []types.BlockID{}
+		if votes == nil || len(votes) == 0 { // if no votes, vote as the whole layer
+			t.Info("Votes is nil. Reading the whole layer")
+			l, e := t.orphans.GetLayer(id)
```

Comment:
continuing the previous comment, why `orphans` ?

Reply:
its just the variable name, the method is correct...

Reply:
Yes. Done. Called it meshProvider. Let me know if it bothers you.